query_optimizer/ExecutionGenerator.cpp

```diff
@@ -563,12 +572,33 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   std::vector<attribute_id> probe_attribute_ids;
   std::vector<attribute_id> build_attribute_ids;
 
+  std::vector<attribute_id> probe_original_attribute_ids;
+  std::vector<attribute_id> build_original_attribute_ids;
+
+  relation_id probe_relation_id;
+  relation_id build_relation_id;
+
   bool any_probe_attributes_nullable = false;
   bool any_build_attributes_nullable = false;
 
+  bool skip_hash_join_optimization = false;
+
   const std::vector<E::AttributeReferencePtr> &left_join_attributes =
       physical_plan->left_join_attributes();
   for (const E::AttributeReferencePtr &left_join_attribute : left_join_attributes) {
+    // Try to determine the original relation referenced in the Hash Join.
+    const CatalogRelation *catalogRelation =
```

Comment:
Variable names are usually ``catalog_relation`` instead of ``catalogRelation``. Also, can you think of a more informative name for this variable?